dbusmock/testcase.py

```diff
@@ -113,15 +113,17 @@ def start_session_bus(cls) -> None:
 
         This gets stopped automatically at class teardown.
         '''
-        DBusTestCase.__start_bus('session')
+        assert cls.session_bus_pid is None
+        cls.__start_bus('session')
```

Comment:
Starting a bus per (derived) class feels oddly in the middle between concepts: Either every test needs its own bus (and then this shouldn't be a classmethod - but that feels a bit excessive), or all test classes are fine with sharing a single bus; the latter was actually the intention, as this starting/stopping happens per test file. Your assertion also seems to support this. 

So I'd actually be in favor of fixing it the other way round: Leaving start_ as it is (global singleton), and fixing the teardown instead. WDYT?

Reply:
I'm in two minds about this. And there are two independent issues with this, I think.

A global singleton would be simpler but this is a testcase class wrapper so I'd guess you want sets of tests (i.e. a class) isolated from each other and not interfere with other sets. Right now if you run test classes in parallel in the same process one class's start/stop would override the other class' bus (Not sure it will actually happen and how you'd trigger this in the same process). But if everything uses the same singleton it can happen. Plus you overwrite the pid so you can't actually stop andy but the most recently started bus. So that would favour the per-class bus.

However: we rely on `setenv`/`getenv` so some  interference and race condition is somewhat guaranteed anyway. Then again that's only on startup and only a narrow window compared to the start/stop sequence.

Functionally, nothing actually changes here because the way unittest and pytest call the fixtures you'll always call `tearDownClass()` first which will stop the current bus. So I think it doesn't matter where the bus is attached to.

Lots of words to say :shrug:, dunno. IMO the right way here is to push consumers towards pytest fixtures instead :)

Reply:
Good point about the tearDownClass(). Let's keep this approach then.